### DISCUSSION TITLE
Hiding/transforming editor characters

### DISCUSSION THREAD

**Discussion by jesnor** [microsoft/vscode-discussions#290] [Extension Development QnA] [answered]
I'm interested in writing a VS Code extension that "prettifies" and hides the visual noise of Rust source code. Some features would be:

- Don't show semicolons and commas at end of rows and automatically add them if they are missing
- Put extra space before things like function call left parenthesis, method calls, array indexing and generic type arguments
- Potentially hide parenthesis, curly braces and <> around simple, single values

For example this Rust code:

```
struct Point {
    x: i32,
    y: i32
}

fn foo(v: Vec<i32>) {
    let x = 20;
    
    if x > 10 {
        v.push(x);
    } else {
        println!("Not big enough!");
    }
}
```

could look like this:

```
struct Point {
    x: i32
    y: i32
}

fn foo (v: Vec i32) {
    let x = 20
    
    if x > 10
        v .push x
    else
        println! "Not big enough!"
}

```

Which I find way easier on the eyes.

It should be possible bind a key to toggle between normal Rust code view and this prettified view. Row and column positions should be the same for characters visible in both views.

I've made some success just by using `"highlight.regexes"` with decorations in my `settings.json`, but it's a bit flakey so I would rather create an extension. The problem is I've never created a VS Code extension before so any help on how to do this and what can be done, and also pointers to documentation/code that might be relevant for this task would be really appreciated!

**Comment by isidorn** [comment, marked as answer]
https://code.visualstudio.com/api
https://code.visualstudio.com/api/get-started/your-first-extension

And I also noticed that [Chat GPT bot](https://chat.openai.com/chat) is pretty good at scaffolding extensions. So you can try interacting with the bot for help.
